test/e2e/kubevirt.go

```diff
@@ -353,24 +366,48 @@ var _ = Describe("Kubevirt Virtual Machines", func() {
 			}
 		}
 
-		startNorthSouthIngressIperfTraffic = func(containerName string, addresses []string, port int32, stage string) error {
+		startNorthSouthIperfTraffic = func(execFn func(cmd string) (string, error), addresses []string, port int32, logPrefix, stage string) error {
 			GinkgoHelper()
 			Expect(addresses).NotTo(BeEmpty())
 			for _, address := range addresses {
-				iperfLogFile := fmt.Sprintf("/tmp/ingress_test_%[1]s_%[2]d_iperf3.log", address, port)
-				output, err := runCommand(containerRuntime, "exec", containerName, "bash", "-c", fmt.Sprintf(`
-iperf3 -c %[1]s -p %[2]d
-killall iperf3
-rm -f %[3]s
-iperf3 -t 0 -c %[1]s -p %[2]d --logfile %[3]s &
-`, address, port, iperfLogFile))
+				iperfLogFile := fmt.Sprintf("/tmp/%s_test_%s_%d_iperf3.log", logPrefix, address, port)
+				By(fmt.Sprintf("remove iperf3 log for %s: %s", address, stage))
+				output, err := execFn(fmt.Sprintf("rm -f %s", iperfLogFile))
 				if err != nil {
-					return fmt.Errorf("%s: %w", output, err)
+					return fmt.Errorf("failed removing iperf3 log file %s: %w", output, err)
+				}
+
+				By(fmt.Sprintf("check iperf3 connectivity for %s: %s", address, stage))
+				output, err = execFn(fmt.Sprintf("iperf3 -c %s -p %d", address, port))
+				if err != nil {
+					return fmt.Errorf("failed checking iperf3 connectivity %s: %w", output, err)
+				}
+
+				By(fmt.Sprintf("start from %s: %s", address, stage))
+				output, err = execFn(fmt.Sprintf("nohup iperf3 -t 0 -c %[1]s -p %[2]d --logfile %[3]s &", address, port, iperfLogFile))
+				if err != nil {
+					return fmt.Errorf("failed at starting iperf3 in background %s: %w", output, err)
 				}
 			}
 			return nil
 		}
 
+		startNorthSouthIngressIperfTraffic = func(containerName string, addresses []string, port int32, stage string) error {
+			GinkgoHelper()
+			execFn := func(cmd string) (string, error) {
+				return runCommand(containerRuntime, "exec", "-i", containerName, "bash", "-c", cmd)
+			}
+			return startNorthSouthIperfTraffic(execFn, addresses, port, "ingress", stage)
+		}
+
+		startNorthSouthEgressIperfTraffic = func(vmi *kubevirtv1.VirtualMachineInstance, addresses []string, stage string) error {
+			GinkgoHelper()
+			execFn := func(cmd string) (string, error) {
+				return kubevirt.RunCommand(vmi, cmd, 5*time.Second)
+			}
+			return startNorthSouthIperfTraffic(execFn, addresses, 5201, "egress", stage)
```

Comment:
maybe we should *not* hardcode the port number ? i.e. follow-up executions of failures will fail if this hangs ... 

And we run tests that fail twice (i.e. the second execution would surely fail if this had hanged.